config/opal_mca.m4

```diff
@@ -720,27 +759,31 @@ AC_DEFUN([MCA_COMPONENT_COMPILE_MODE],[
     # there is a tie (either neither or both specified), prefer
     # static.
     if test "$STATIC_COMPONENT" = "1"; then
-        $4=static
+        compile_mode_internal_tmp=static
     elif test "$SHARED_COMPONENT" = "1"; then
-        $4=dso
+        compile_mode_internal_tmp=dso
     elif test "$STATIC_FRAMEWORK" = "1"; then
-        $4=static
+        compile_mode_internal_tmp=static
     elif test "$SHARED_FRAMEWORK" = "1"; then
-        $4=dso
+        compile_mode_internal_tmp=dso
     elif test "$STATIC_all" = "1"; then
-        $4=static
+        compile_mode_internal_tmp=static
     elif test "$DSO_all" = "1"; then
-        $4=dso
+        compile_mode_internal_tmp=dso
     else
-        $4=static
+        compile_mode_internal_tmp=static
     fi
 
+    AS_VAR_SET([$4], [$compile_mode_internal_tmp])
+
     AC_MSG_CHECKING([for MCA component $2:$3 compile mode])
     if test "$DIRECT_$2" = "$3" ; then
-        AC_MSG_RESULT([$$4 - direct])
+        AC_MSG_RESULT([$compile_mode_internal_tmp - direct])
     else
-        AC_MSG_RESULT([$$4])
+        AC_MSG_RESULT([$compile_mode_internal_tmp])
     fi
+
+    OPAL_VAR_SCOPE_POP([compile_mode_internal_tmp])
```

Comment:
`OPAL_VAR_SCOPE_POP` doesn't take any arguments -- it unwinds *all* the variable names from the most recent `OPAL_VAR_SCOPE_PUSH`.